app/scripts/controllers/permissions/specifications.js

```diff
@@ -35,6 +35,11 @@ const CaveatFactories = Object.freeze({
   [CaveatTypes.restrictReturnedAccounts]: (accounts) => {
     return { type: CaveatTypes.restrictReturnedAccounts, value: accounts };
   },
+
+  [CaveatTypes.restrictNetworkSwitching]: (network) => {
+    console.log('caveat factory function', network);
+    return { type: CaveatTypes.restrictNetworkSwitching, value: network };
```

Comment:
Should this be `network` singular or `networks` plural? I would also suggest: `chainIds`.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/24415/commits/47967890ab0e0efe9640adc6c5fe982693937598